nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -74,6 +77,43 @@ public static DocumentModel moveContentToColdStorage(CoreSession session, Docume
         return session.saveDocument(documentModel);
     }
 
+    /**
+     * Retrieves the cold storage content associated with the document of the given {@link DocumentRef}.
+     * 
+     * @param session the core session
+     * @param documentRef the document reference
+     * @param numberOfDaysOfAvailability number of days that you want your cold storage content to be accessible after
+     *            restoring
+     * @apiNote This method will initiate a request restore, calling the {@link Blob#getStream()} during this process
+     *          doesn't mean you will get the blob content immediately.
+     * @return the updated document model if the retrieve succeeds
+     * @throws NuxeoException if there is no cold storage content associated with the given document, or if it is being
+     *             retrieved
+     */
+    public static DocumentModel retrieveContentFromColdStorage(CoreSession session, DocumentRef documentRef,
+            int numberOfDaysOfAvailability) {
+        DocumentModel documentModel = session.getDocument(documentRef);
+        log.debug("Retrieve from cold storage the content of document: {} for: {} days", () -> documentModel,
+                () -> numberOfDaysOfAvailability);
+
+        if (!documentModel.hasFacet(FacetNames.COLD_STORAGE)
+                || documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY) == null) {
+            throw new NuxeoException(String.format("No cold storage content defined for document: %s.", documentModel),
+                    SC_NOT_FOUND);
+        }
+
+        Serializable beingRetrieved = documentModel.getPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY);
+        if (Boolean.TRUE.equals(beingRetrieved)) {
+            throw new NuxeoException(
+                    String.format("The cold storage content associated with the document: %s is being retrieved.",
+                            documentModel),
+                    SC_CONFLICT);
+        }
+
+        documentModel.setPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY, true);
+        return session.saveDocument(documentModel);
```

Comment:
I wonder if we want/need to store the numberOfDaysOfAvailability, or if this information will come from the cold storage when checking for availability, see https://github.com/nuxeo/nuxeo/pull/3666/commits/aa516a7728bc3b0ff6adf1b73cbe788845a41f8e#diff-ead64399e0b98c4a84e7b8cef49b5444R169, in which case it can then be passed dynamically to the fired event.

Reply:
i agree i asked myself the same question, i think we will store when the content is really available and this information will be in the mail. but as i don't know how to get this information i didn't do it perhaps i will add a `TODO` wdyt

Reply:
yep, go for the TODO